aicsimageio/readers/tiff_reader.py

```diff
@@ -178,80 +185,100 @@ def _read_immediate(self) -> np.ndarray:
-                    # Add scene dimension only if there are multiple scenes
-                    if len(tiff.series) == 1:
-                        self._dims = single_scene_dims
-                    else:
-                        self._dims = f"{Dimensions.Scene}{single_scene_dims}"
-                # Sometimes the dimension info is wrong in certain dimensions, so guess
-                # that dimension
-                else:
-                    guess = self.guess_dim_order(tiff.series[0].pages.shape)
-                    best_guess = []
-                    for dim_from_meta in single_scene_dims:
-                        if dim_from_meta in Dimensions.DefaultOrder:
-                            best_guess.append(dim_from_meta)
-                        else:
-                            appended_dim = False
-                            for guessed_dim in guess:
-                                if guessed_dim not in best_guess:
-                                    best_guess.append(guessed_dim)
-                                    appended_dim = True
-                                    log.info(
-                                        f"Unsure how to handle dimension: "
-                                        f"{dim_from_meta}. "
-                                        f"Replaced with guess: {guessed_dim}"
-                                    )
-                                    break
-
-                            # All of our guess dims were already in the dim list,
-                            # append the dim read from meta
-                            if not appended_dim:
-                                best_guess.append(dim_from_meta)
-
-                    best_guess = "".join(best_guess)
-
-                    # Add scene dimension only if there are multiple scenes
-                    if len(tiff.series) == 1:
-                        self._dims = best_guess
-                    else:
-                        self._dims = f"{Dimensions.Scene}{best_guess}"
+            props = self._get_tiff_properties(self._file, self.specific_s_index)
+            self._dims = props.dims
+            self._shape = props.shape
+            self._dtype = props.dtype
 
         return self._dims
 
-    @dims.setter
-    def dims(self, dims: str):
-        # Check amount of provided dims against data shape
-        if len(dims) != len(self.dask_data.shape):
-            raise exceptions.InvalidDimensionOrderingError(
-                f"Provided too many dimensions for the associated file. "
-                f"Received {len(dims)} dimensions [dims: {dims}] "
-                f"for image with {len(self.data.shape)} dimensions "
-                f"[shape: {self.data.shape}]."
-            )
-
-        # Set the dims
-        self._dims = dims
+    @property
+    def shape(self) -> Tuple[int]:
+        if self._shape is None:
+            props = self._get_tiff_properties(self._file, self.specific_s_index)
+            self._dims = props.dims
```

Comment:
is it ever possible that `_shape is None` but _dims or _dtype was previously set?

Reply:
GOOD CATCH. `dims` can be set with the new setter.

Reply:
After discussion with @toloudis, I have actually decided to remove dimension setting entirely from the base readers. You can still inform the `AICSImage` object of custom dimensions with `AICSImage(..., known_dims="BLAH")` but the _reader_ objects are to some level the raw data (even with our best guessing on certain readers)